### PR TITLE
feat(range-server): integrate range offload progress to metadata manager

### PR DESCRIPTION
**Pull request by superhx**

### CODE REVIEW

File: components/object-storage/src/lib.rs [hunk 1]
```diff
@@ -21,6 +21,9 @@ use mockall::automock;
 
 use tokio::sync::{broadcast, mpsc};
 
+pub type OffloadProgress = Vec<((u64, u32), u64)>;
```
Comment (Chillax-0v0) [inline components/object-storage/src/lib.rs:24]: Why not use `RangeKey` here?

File: components/object-storage/src/object_manager.rs [hunk 2]
```diff
@@ -179,33 +180,29 @@ impl OffloadingObjects {
...
+    /// Listeners of range offload progress.
+    offload_progress_listeners: Vec<mpsc::UnboundedSender<OffloadProgress>>,
 }
 
 impl Metadata {
     /// Save object metadata.
-    /// If `need_flush` is true, flush the offload offset of the range.
-    fn add_object(&mut self, object: &ObjectMetadata, need_flush: bool) {
+    /// return offload offset.
+    fn add_object(&mut self, object: &ObjectMetadata) -> Option<u64> {
```
Comment (Chillax-0v0) [inline components/object-storage/src/object_manager.rs:200]: When will this func return a `None`?

Comment (superhx) [reply components/object-storage/src/object_manager.rs:200]: When the range isn't held by current range server, we don't need the offload offset.

File: components/object-storage/src/lib.rs (hunk 1 above)
Comment (superhx) [reply components/object-storage/src/lib.rs:24]: I will unify RangeKey to range/mod.rs Range type in later PR.

**Review by Chillax-0v0 — APPROVED** [review]

**Pull request merged**